File: study3_200218/jiho/answer1.py
def answer1(words):
    max_length = max([len(word) for word in words])

    ret = []
    for c in range(max_length):
        for r in range(len(words)):
            if len(words[r]) > c:
                ret.append(words[r][c])

    return "".join(ret)


print(answer1(["ABCDE", "abcde", "01234", "FGHIJ", "fghij",]))

File: study3_200218/jiho/answer1.js
const answer1 = words => {
  maxLength = Math.max(...words.map(w => w.length));
  result = [];
  for (let c = 0; c < maxLength; c++) {
    for (let r = 0; r < 5; r++) {
      if (c < words[r].length) {
        result.push(words[r][c]);
      }
    }
  }
  return result.join("");
};

console.log(answer1(["ABCDE", "abcde", "01234", "FGHIJ", "fghij"]));
console.log(answer1(["AABCDD", "afzz", "09121", "a8EWg6", "P5h3kx"]));
